File: LP2SoftRRHHModel/src/pe/edu/pucp/lp2soft/rrhh/model/Empleado.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.edu.pucp.lp2soft.rrhh.model;

import java.util.Date;

/**
 *
 * @author axeli
 */
public class Empleado extends Persona {
    private String cargo ; 
    private double sueldo ; 
    private boolean activo ; 
    private Area area ;
    private CuentaUsuario cuentaUsuario ;

    public Empleado (int idPersona, String DNI, String nombre, String apellidoPaterno, char genero, Date fechaNacimiento ,String cargo, double sueldo, Area area) {
        super(idPersona, DNI, nombre, apellidoPaterno, genero, fechaNacimiento);
        this.cargo = cargo;
        this.sueldo = sueldo;
        this.area = area;
    }

    
    
    
    public String getCargo() {
        return cargo;
    }

    public void setCargo(String cargo) {
        this.cargo = cargo;
    }

    public double getSueldo() {
        return sueldo;
    }

    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }

    public boolean isActivo() {
        return activo;
    }

    public void setActivo(boolean activo) {
        this.activo = activo;
    }

    public Area getArea() {
        return area;
    }

    public void setArea(Area area) {
        this.area = area;
    }

    public CuentaUsuario getCuentaUsuario() {
        return cuentaUsuario;
    }

    public void setCuentaUsuario(CuentaUsuario cuentaUsuario) {
        this.cuentaUsuario = cuentaUsuario;
    }
    
    
}

File: ProyectosNET/LP2Soft/LP2SoftRRHHModel/Empleado.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LP2SoftRRHHModel
{
    public class Empleado : Persona
    {
        private string _cargo;
        private double _sueldo;
        private bool _activo;
        private CuentaUsuario _cuentaUsuario;
        private Area _area;

        public Empleado() { }

        public Empleado(string DNI, string nombre, string apellidoPaterno, char genero, DateTime fechaNacimiento, 
            string cargo, double sueldo) : base(DNI,nombre,apellidoPaterno,genero,fechaNacimiento)
        {
            _cargo = cargo;
            _sueldo = sueldo;
        }

        public string Cargo { get { return _cargo;  } set => _cargo = value; }
        public double Sueldo { get => _sueldo; set => _sueldo = value; }
        public bool Activo { get => _activo; set => _activo = value; }
        public CuentaUsuario CuentaUsuario { get => _cuentaUsuario; set => _cuentaUsuario = value; }
        public Area Area { get => _area; set => _area = value; }
    }
}
